execute/tokendata/usdc/usdc.go

```diff
@@ -4,43 +4,168 @@ import (
 	"context"
 	"fmt"
 
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"golang.org/x/exp/maps"
+
 	"github.com/smartcontractkit/chainlink-ccip/execute/exectypes"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
 	"github.com/smartcontractkit/chainlink-ccip/pluginconfig"
 )
 
 type USDCCCTPTokenDataObserver struct {
-	configs []pluginconfig.USDCCCTPObserverConfig
+	configs           pluginconfig.USDCCCTPObserverConfig
+	supportedTokens   map[string]struct{}
+	usdcMessageReader reader.USDCMessageReader
+	attestationClient AttestationClient
+}
+
+func NewUSDCCCTPTokenDataObserver(configs pluginconfig.USDCCCTPObserverConfig) *USDCCCTPTokenDataObserver {
+	return NewUSDCCCTP(
+		configs,
+		nil,
+		nil,
+	)
+}
+
+func NewUSDCCCTP(
```

Comment:
Both this and the above are public facing constructors, lets have either some docs on which one to use for which scenario or just have one `NewTokenDataObserver` (since package name is usdc, it reads like `usdc.NewTokenDataObserver`) to reduce likelihood of using wrong one and confusion